addons/sale_project/tests/test_sale_project.py

```diff
@@ -160,3 +160,30 @@ def test_sol_product_type_update(self):
 
         self.product_order_service3.type = 'service'
         self.assertTrue(sale_order_line.is_service, "As the product is a service, the SOL should be a service")
+
+    @users('demo')
+    def test_cancel_so_linked_to_project(self):
+        """ Test that cancelling a SO linked to a project will not raise an error """
+        # Ensure user don't have edit right access to the project
+        group_sale_manager = self.env.ref('sales_team.group_sale_manager')
+        group_project_user = self.env.ref('project.group_project_user')
+        self.env.user.write({'groups_id': [(6, 0, [group_sale_manager.id, group_project_user.id])]})
+
+        partner = self.env['res.partner'].create({'name': "Mur en brique"})
+        sale_order = self.env['sale.order'].with_context(tracking_disable=True).create({
+            'partner_id': partner.id,
+            'partner_invoice_id': partner.id,
+            'partner_shipping_id': partner.id,
+            'project_id': self.project_global.id,
+        })
+        self.env['sale.order.line'].create({
+            'name': self.product_order_service2.name,
+            'product_id': self.product_order_service2.id,
+            'product_uom_qty': 10,
+            'product_uom': self.product_order_service2.uom_id.id,
+            'price_unit': self.product_order_service2.list_price,
+            'order_id': sale_order.id,
+        })
+        sale_order.action_confirm()
+        # The test is here: canceling order should not raise an access error when removing sale_line_id from project
+        sale_order.action_cancel()
```

Comment:
You could also check if the project has no SOL set on its `sale_line_id` field. 🙂

Reply:
Done :+1: However, I checked on the `project.tasks` because `sale_line_id` on project_id seems to be only manually filled.